### PR TITLE
Follow up insights - user metrics

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Add single team members bookings that don't belong to team to any metric.

https://www.loom.com/share/30c75cb2e5234b2dac6fc0af896d1bfa

**Environment**: Staging(main branch) 

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Move to packages/prisma, run yarn seed-insights couple times.
- [ ] Enter with desired user member or admin account.
- [ ] Use filters and see numbers change.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

Validate we have everything in this comment:
https://github.com/calcom/cal.com/pull/7922#issuecomment-1483131765

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]

File: apps/web/pages/insights/index.tsx [hunk 1]
```diff
@@ -99,7 +94,7 @@ export default function InsightsPage() {
                   <LeastBookedTeamMembersTable />
                 </div>
                 <small className="block text-center text-gray-600">
-                  {t("looking_for_more_analytics")}
+                  {t("looking_for_more_insights")}
```
Comment (alannnc) [inline apps/web/pages/insights/index.tsx:97]: Style or text changes.

File: packages/features/insights/components/AverageEventDurationChart.tsx [hunk 2]
```diff
@@ -6,34 +6,45 @@ import { trpc } from "@calcom/trpc";
...
+      {data && data.length > 0 && !isNoData && (
+        <LineChart
+          className="mt-4 h-80"
+          data={data}
+          index="Date"
+          categories={["Average"]}
+          colors={["blue"]}
+          valueFormatter={valueFormatter}
+        />
+      )}
```
Comment (alannnc) [inline packages/features/insights/components/AverageEventDurationChart.tsx:47]: Added logic components to display if data is empty.

File: packages/features/insights/components/LoadingInsights.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        <svg
+          className={classNames("mx-4 h-8 w-8 animate-spin", "text-black dark:text-white")}
+          xmlns="http://www.w3.org/2000/svg"
+          fill="none"
+          viewBox="0 0 24 24">
+          <circle className="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4" />
+          <path
+            className="opacity-75"
+            fill="currentColor"
+            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
```
Comment (alannnc) [inline packages/features/insights/components/LoadingInsights.tsx:20]: Component that can be reused inside every card from insights while loading.

File: packages/features/insights/context/FiltersProvider.tsx [hunk 4]
```diff
@@ -128,6 +157,30 @@ export function FiltersProvider({ children }: { children: React.ReactNode }) {
...
+          if ((teamId && !userId) || (userId && teamId)) {
+            query.teamId = parsedTeamId;
+            setSelectedTeamId(parsedTeamId);
+            setSelectedUserId(null);
+          } else if (userId && !teamId) {
+            query.userId = parsedUserId;
+            setSelectedUserId(parsedUserId);
+            setSelectedTeamId(null);
+          }
+
```
Comment (alannnc) [inline packages/features/insights/context/FiltersProvider.tsx:179]: Cleaning url and context state.

File: packages/prisma/seed-insights.ts [hunk 5]
```diff
@@ -5,6 +5,65 @@ import { v4 as uuidv4 } from "uuid";
 import dayjs from "@calcom/dayjs";
 import { hashPassword } from "@calcom/features/auth/lib/hashPassword";
 
+const shuffle = (
```
Comment (alannnc) [inline packages/prisma/seed-insights.ts:8]: Moved to a reusable function.

File: packages/prisma/seed-insights.ts [hunk 6]
```diff
@@ -75,44 +143,146 @@ async function main() {
...
 
-  let teamEvents = await prisma.eventType.findMany({
+  const userSingleEvents = await prisma.eventType.findMany({
     where: {
-      teamId: insightsTeam.id,
+      userId: insightsUser.id,
     },
   });
 
+  if (userSingleEventsAdmin.length === 0 && userSingleEvents.length === 0) {
```
Comment (alannnc) [inline packages/prisma/seed-insights.ts:192]: I added bookins and event types that belong to team members but they don't belong to the team.

**Pull request merged**